### PR TITLE
set UCX_TLS in impi with UCX

### PR DESCRIPTION
**Pull request by lexming**
Fixes issue https://github.com/easybuilders/easybuild-easyconfigs/issues/10899

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _impi-2019.7.217-iccifort-2020.1.217.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node303.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/c632e648f6662279ddcd11e36bb6b717 for a full test report.

File: easybuild/easyblocks/i/impi.py [hunk 1]
```diff
@@ -313,4 +314,11 @@ def make_module_extra(self, *args, **kwargs):
             # -fc also works, but -f90 takes precedence
             txt += self.module_generator.set_alias('mpiifort', 'mpiifort -f90=ifort')
 
+        # set environment variable UCX_TLS to 'all', this will work in all setups
+        # needed with UCX regardless of the transports available (even without a Mellanox HCA)
+        # since impi v2019.8, the MLX provider works without UCX_TLS, but setting it does not hurt
+        ucx_root = get_software_root('UCX')
+        if ucx_root:
+            txt += self.module_generator.set_environment('UCX_TLS', 'all')
```
Comment (akesandgren) [inline easybuild/easyblocks/i/impi.py:322]: Are you sure that all versions of UCX up until 1.9.0 knows about "all" ?

**Comment by lexming** [comment]
Test report by @lexming
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _impi-2019.8.254-iccifort-2020.1.217.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node303.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/98af5568335b5f3505f6eef578ece658 for a full test report.

**Comment by lexming** [comment]
For reference, previous test is in a machine without a Mellanox HCA (so no MLX provider). The same test with current `impi` easyblock in `develop` fails the sanity checks. See https://gist.github.com/lexming/4e64523090ddf1c48baf4b47353e918b

File: easybuild/easyblocks/i/impi.py (hunk 1 above)
Comment (lexming) [reply easybuild/easyblocks/i/impi.py:322]: yes, added ages ago with https://github.com/openucx/ucx/pull/104

Comment (akesandgren) [reply easybuild/easyblocks/i/impi.py:322]: ok

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @lexming!

**Comment by lexming** [comment]
Test report by @lexming
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _impi-2019.9.304-iccifort-2020.4.304.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node303.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/a466d5bd8b549ef4d102f23459cb1630 for a full test report.

File: easybuild/easyblocks/i/impi.py (hunk 1 above)
Comment (lexming) [reply easybuild/easyblocks/i/impi.py:322]: > also what is the difference between UCX_TLS=all and not setting UCX_TLS? Does UCX_TLS=all include more transports?

There is no difference, in both cases UCX will consider all available TLS and choose the best one. So explicitly setting `UCX_TLS=all` will not change the behaviour of systems that were already working well without it.

**Pull request merged**